### PR TITLE
feat: migrate topic responses to new pattern

### PR DESCRIPTION
**Pull request by rishtigupta**
- Migrate TopicPublish and TopicSubscribe to new pattern

### CODE REVIEW

File: packages/core/src/messages/responses/topic-publish.ts [hunk 1]
```diff
@@ -1,38 +1,16 @@
...
- *   // `TopicPublish.Error` in this block, so you will have access to the properties
- *   // of the Error class; e.g. `response.errorCode()`.
- * }
- * ```
- */
-export abstract class Response extends ResponseBase {}
-
-class _Success extends Response {}
+interface IResponse {
+  type: TopicPublishResponse;
```
Comment (anitarua) [inline packages/core/src/messages/responses/topic-publish.ts:6]:
```suggestion
  readonly type: TopicPublishResponse;
```

File: packages/core/src/messages/responses/topic-subscribe.ts [hunk 2]
```diff
@@ -1,28 +1,11 @@
...
- * if (response instanceof TopicSubscribe.Error) {
- *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
- *   // `CacheGet.Error` in this block, so you will have access to the properties
- *   // of the Error class; e.g. `response.errorCode()`.
- * }
- * ```
- */
-export abstract class Response extends ResponseBase {}
+interface IResponse {
+  type: TopicSubscribeResponse;
```
Comment (anitarua) [inline packages/core/src/messages/responses/topic-subscribe.ts:7]:
```suggestion
  readonly type: TopicSubscribeResponse;
```

**Review by anitarua — COMMENTED** [review]
looks good overall!

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

**Pull request merged**